### PR TITLE
feat(query): enable min/max pushdown

### LINKED ISSUE #21155: Enable group min/max pushdown [backport 2.0.x]
Backport #20994 to the `2.0` branch

### PR DESCRIPTION
**Pull request by dgnorton**
Closes #21155 

This PR backports the min/max pushdown to the `2.0` branch for release in 2.0.5.

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -47,7 +47,7 @@ require (
 	github.com/hashicorp/vault/api v1.0.2
 	github.com/imdario/mergo v0.3.9 // indirect
 	github.com/influxdata/cron v0.0.0-20191203200038-ded12750aac6
-	github.com/influxdata/flux v0.112.1
+	github.com/influxdata/flux v0.113.0
```
Comment (danxmoran) [inline go.mod:50]: There's a line in `CHANGELOG.md` for the upgrade to `v0.112.1`, can you update it to ref `v0.113.0` and this PR instead?

File: pkger/service.go [hunk 2]
```diff
@@ -936,7 +936,7 @@ type ImpactSummary struct {
 	Summary Summary
 }
 
-var reCommunityTemplatesValidAddr = regexp.MustCompile(`(?:https://raw.githubusercontent.com/influxdata/community-templates/master/)(?P<name>\w+)(?:/.*)`)
+var reCommunityTemplatesValidAddr = regexp.MustCompile(`(?:https://raw\.githubusercontent\.com/influxdata/community-templates/master/)(?P<name>\w+)(?:/.*)`)
```
Comment (danxmoran) [inline pkger/service.go:939]: Should this one be `CHANGELOG`'d too?

File: v1/services/storage/store.go [hunk 3]
```diff
@@ -208,7 +208,11 @@ func (s *Store) ReadGroup(ctx context.Context, req *datatypes.ReadGroupRequest)
 		return nil, err
 	}
 
-	shardIDs, err := s.findShardIDs(database, rp, false, start, end)
+	// Due to some optimizations around how flux's `last()` function is implemented with the
+	// storage engine, we need to detect if the read request requires a descending
+	// cursor or not.
+	descending := !reads.IsAscendingGroupAggregate(req)
+	shardIDs, err := s.findShardIDs(database, rp, descending, start, end)
```
Comment (danxmoran) [inline v1/services/storage/store.go:215]: The `last()` bug strikes again!

File: ui/package.json [hunk 4]
```diff
@@ -135,7 +135,7 @@
   "dependencies": {
     "@influxdata/clockface": "2.3.4",
     "@influxdata/flux": "^0.5.1",
-    "@influxdata/flux-lsp-browser": "^0.5.31",
+    "@influxdata/flux-lsp-browser": "^0.5.41",
```
Comment (danxmoran) [inline ui/package.json:138]: Meta: I don't think these changes actually do anything now that we pull from the `ui` repo, we should tell the Flux team they don't need to bump this any more (unless it's needed for `e2e` to pass?)

@wbaker85 can you make sure our tag in the UI repo includes this newest version?

File: go.mod (hunk 1 above)
Comment (dgnorton) [reply go.mod:50]: done

File: pkger/service.go (hunk 2 above)
Comment (dgnorton) [reply pkger/service.go:939]: done

File: v1/services/storage/store.go (hunk 3 above)
Comment (dgnorton) [reply v1/services/storage/store.go:215]: @danxmoran is there something we need to change?

Comment (danxmoran) [reply v1/services/storage/store.go:215]: No, was just surprised to see this bug pop up in a 2nd place.

File: storage/flux/table_test.go [hunk 5]
```diff
@@ -24,7 +24,7 @@ import (
 	"github.com/influxdata/flux/memory"
 	"github.com/influxdata/flux/plan"
 	"github.com/influxdata/flux/values"
-	"github.com/influxdata/influxdb/v2"
+	"github.com/influxdata/influxdb"
```
Comment (danxmoran) [inline storage/flux/table_test.go:27]: I think the build failures are because of this change

File: ui/package.json (hunk 4 above)
Comment (wbaker85) [reply ui/package.json:138]: UI repo release for 2.0.5 is updated with the newest flux version!

As far as if the updates are needed for our `e2e` tests to pass...I *doubt* it, but I don't know for sure. I'm going to really push for getting the `e2e` testing moved over to `monitor-ci` in the next few days so at that point we can definitely stop updating `flux-lsp-browser` here (and just delete all the UI source entirely).

**Review by danxmoran — APPROVED** [review]

**Pull request merged**